--- src/Intervention/Image/Gd/Shapes/PolygonShape.php	2025-07-09 09:29:58.026270951 +1000
+++ src/Intervention/Image/Gd/Shapes/PolygonShape.php	2025-07-09 09:30:34.940638065 +1000
@@ -36,12 +36,12 @@
     public function applyToImage(Image $image, $x = 0, $y = 0)
     {
         $background = new Color($this->background);
-        imagefilledpolygon($image->getCore(), $this->points, intval(count($this->points) / 2), $background->getInt());
+        imagefilledpolygon($image->getCore(), $this->points, $background->getInt());
         
         if ($this->hasBorder()) {
             $border_color = new Color($this->border_color);
             imagesetthickness($image->getCore(), $this->border_width);
-            imagepolygon($image->getCore(), $this->points, intval(count($this->points) / 2), $border_color->getInt());
+            imagepolygon($image->getCore(), $this->points, $border_color->getInt());
         }
     
         return true;
